File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
     int l = nums.length;
     int zeroCount = 0;
     int zeroIndex = -1;
     int maxMulti = 1;
     for (int i = 0; i < l; i++){
        if (nums[i] == 0){
            zeroCount++;
            zeroIndex = i;
        } else {
        maxMulti = maxMulti * nums[i];
        }
    
     }
     if (zeroCount > 1){
        return new int[l];
     }
     if (zeroCount == 1){
        int[] arr = new int[l];
        arr[zeroIndex] = maxMulti;
        return arr;
     }
     int[] arr = new int[l];
     for (int i = 0; i < l; i++){
        arr[i] = maxMulti/nums[i];
     }
     return arr;

    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.ts
function productExceptSelf(nums: number[]): number[] {
    let zeroCount = 0
    let total = 1
    let zeroIdx = -1
    for (let i = 0; i < nums.length; i++){
        if (nums[i]===0){
            zeroCount++
            zeroIdx = i
        }
    if (nums[i]!=0){
        total = total * nums[i]
    }
    }
    if (zeroCount >= 2){
        return Array(nums.length).fill(0)
    } else if (zeroCount == 1){
        let ans = new Array(nums.length).fill(0)
        ans[zeroIdx] = total
        return ans
    } else {
        for (let i = 0; i < nums.length; i++){
            nums[i] = total/ nums[i]
        }
    }
    return nums
};